examples/nested_requests.rs

```diff
@@ -0,0 +1,160 @@
+use nvim_rs::{
+  compat::tokio::Compat, create::tokio as create, neovim::Neovim, Handler,
+};
+
+use async_trait::async_trait;
+
+use rmpv::Value;
+
+use std::sync::Arc;
+
+use tokio::{
+  self,
+  process::{ChildStdin, Command},
+  spawn,
+};
+
+use futures::lock::Mutex;
+use futures::select;
+use futures::future::FutureExt;
+
+const NVIMPATH: &str = "neovim/build/bin/nvim";
+
+#[derive(Clone)]
+struct NeovimHandler {
+  froodle: Arc<Mutex<String>>,
+}
+
+#[async_trait]
+impl Handler for NeovimHandler {
+  type Writer = Compat<ChildStdin>;
+
+  async fn handle_request(
+    &self,
+    name: String,
+    args: Vec<Value>,
+    neovim: Neovim<Compat<ChildStdin>>,
+  ) -> Result<Value, Value> {
+    match name.as_ref() {
+      "dummy" => Ok(Value::from("o")),
+      "req" => {
+        let v = args[0].as_str().unwrap();
+
+        let neovim = neovim.clone();
+        match v {
+          "y" => {
+            let mut x: String = neovim
+              .get_vvar("progname")
+              .await
+              .unwrap()
+              .as_str()
+              .unwrap()
+              .into();
+            x.push_str(" - ");
+            x.push_str(
+              neovim.get_var("oogle").await.unwrap().as_str().unwrap(),
+            );
+            x.push_str(" - ");
+            x.push_str(
+              neovim
+                .eval("rpcrequest(1,'dummy')")
+                .await
+                .unwrap()
+                .as_str()
+                .unwrap(),
+            );
+            x.push_str(" - ");
+            x.push_str(
+              neovim
+                .eval("rpcrequest(1,'req', 'z')")
+                .await
+                .unwrap()
+                .as_str()
+                .unwrap(),
+            );
+            Ok(Value::from(x))
+          }
+          "z" => {
+            let x: String = neovim
+              .get_vvar("progname")
+              .await
+              .unwrap()
+              .as_str()
+              .unwrap()
+              .into();
+            Ok(Value::from(x))
+          }
+          &_ => Err(Value::from("wrong argument to req")),
+        }
+      }
+      &_ => Err(Value::from("wrong method name for request")),
+    }
+  }
+
+  async fn handle_notify(
+    &self,
+    name: String,
+    args: Vec<Value>,
+    _neovim: Neovim<Compat<ChildStdin>>,
+  ) {
+    match name.as_ref() {
+      "set_froodle" => {
+        *self.froodle.lock().await = args[0].as_str().unwrap().to_string()
+      }
+      _ => {}
+    };
+  }
+}
+
+#[tokio::main(basic_scheduler)]
```

Comment:
This needs updating, I think `(flavor = "current_thread")` should do it. We definitely want this one to run on just one thread (and analogously the test).